### PR TITLE
Yap Shi Yuan - section 1 - scissors-paper-stone

### PR DESCRIPTION
**Pull request by Yap83**
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment?
Roughly 2 plus to 3 hours

Please fill in one error and/or error message you received while working on this assignment.
extra spacing in the stone causing undefine data output

What part of the assignment did you spend the most time on?
Finding error 

Comfort Level (1-5):
4

Completeness Level (1-5):
5

What did you think of this deliverable?
ok

Is there anything in this code that you feel pleased about?
Math.random as it can create a lot of random number for games

### CODE REVIEW

**Comment by Yap83** [comment]
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment?
Roughly 10 hours for part 2

Please fill in one error and/or error message you received while working on this assignment.
wrong name causing unable to define

What part of the assignment did you spend the most time on?
creating Korean game mode and restructure of program due to creating different mode

Comfort Level (1-5):
4

Completeness Level (1-5):
4.5

What did you think of this deliverable?
ok

Is there anything in this code that you feel pleased about?
global variable help in solving most data collection issue

File: script.js [hunk 1]
```diff
@@ -1,4 +1,654 @@
...
+var noOfGame = 0;
+var userDraw = 0;
+var nickName = " ";
+var gameMode = "normal";
+var playMode = "normal";
+var koreanMode = 0;
+var koreanLoseMode = 0;
+var koreanWinMode = 0;
+var comAScore = 0;
+var comBScore = 0;
```
Comment (eddiejpot) [inline script.js:11]: Good use of global variables!

File: script.js [hunk 2]
```diff
@@ -1,4 +1,654 @@
...
+  if (actualResult == 1) {
+    return "scissors";
+  }
+  if (actualResult == 2) {
+    return "paper";
+  }
+  if (actualResult == 3) {
+    return "stone";
+  }
+};
```
Comment (eddiejpot) [inline script.js:26]: Nice job decomposing the logic out :) !
One note. Our functions should be named **starting with a verb** and also **describe what it does**.

For example this function can be name `getScissorsPaperOrStone `or `getSPS`

File: script.js [hunk 3]
```diff
@@ -1,4 +1,654 @@
...
+  }
+  if (mode == "com" && noOfGame != 0) {
+    gameMode = "com";
+    playMode = gameMode;
+    userScore = 0;
+    noOfGame = 1;
+    comAScore = 0;
+    comBscore = 0;
+    userDraw = 0;
+  }
```
Comment (eddiejpot) [inline script.js:100]: Interesting way to separate the logic out in a function. nice!

File: script.js [hunk 4]
```diff
@@ -1,4 +1,654 @@
...
+        userDraw +
+        "</b>";
+    }
+  } else if (
+    (k2Input1 == "scissors" && k2ComResult1 == "paper") ||
+    (k2Input1 == "paper" && k2ComResult1 == "stone") ||
+    (k2Input1 == "stone" && k2ComResult1 == "scissors") ||
+    (k2Input1 == "paper" && k2ComResult1 == "scissors") ||
+    (k2Input1 == "stone" && k2ComResult1 == "paper") ||
+    (k2Input1 == "scissors" && k2ComResult1 == "stone")
```
Comment (eddiejpot) [inline script.js:469]: Good use of the || operator here!

**Review by eddiejpot — COMMENTED** [review]
Overall great work! You got really far! 

This shows me that you have a good understanding or operators and functions. 
One thing to improve on would be your function naming :) ! 

Naming is pretty important in programming. You would want anyone who reads your code to immediately understand what that function you wrote does

**Pull request closed without merging**